packages/react/src/internal/useNormalizedInputProps.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright IBM Corp. 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  WarningFilled16,
+  WarningAltFilled16,
+  EditOff16,
+} from '@carbon/icons-react';
+import { settings } from 'carbon-components';
+
+const { prefix } = settings;
+
+export function useNormalizedInputProps({
```

Comment:
I think this is a good point of abstraction. Could you add a jsdoc-type comment for it so we have some context on what this hook accomplishes, when it should be used, expected param types, etc?

Reply:
Pushed an update adding some guidance aorund it. Let me know what you think of it!